### PR TITLE
Add compilation to dev task, disable javadoc task for now

### PR DESCRIPTION
**Pull request by mbaxter**
## PR Description
Update gradle dev task to include compilation.  Disable javadoc task for now, since its failing.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,10 +19,16 @@ defaultTasks 'build','checkLicenses'
 
 def buildAliases = [
   'dev': [
+    'compileJava',
+    'compileTestJava',
```
Comment (cemozerr) [inline build.gradle:23]: What do these lines do? from 22 to 26? were we not compiling before?

File: build.gradle [hunk 2]
```diff
@@ -19,10 +19,16 @@ defaultTasks 'build','checkLicenses'
...
+    'compileJmhJava',
+    'compileIntegrationTestJava',
+    'compileReferenceTestJava',
     'spotlessApply',
     'build',
     'checkLicenses',
-    'javadoc'],
+    // TODO - Fix javadoc task
+    /*'javadoc'*/],
```
Comment (cemozerr) [inline build.gradle:31]: That looks good to me.

**Review by ajsutton — APPROVED** [review]
LGTM. Slightly surprised compilation wasn't included in `build` though maybe it missed the reference tests and integrations tests.

File: build.gradle (hunk 1 above)
Comment (mbaxter) [reply build.gradle:23]: At least some of these tasks weren't running before :D I basically want to have a dev task that will catch most simple errors before pushing up a branch and running CI.  So, I just copied some additional tasks from the circleci config to try to ensure that the basic checks were running.

Comment (ajsutton) [reply build.gradle:23]: Purely for knowledge sharing, my habit is to use command-F9 to tell IntelliJ to build everything.  Picks up compilation errors nicely and in the IDE so it's easier for me to fix.

Still definitely worth having a gradle target that does sensible checks though.

**Pull request merged**